tests/unit/test_core.py

```diff
@@ -0,0 +1,85 @@
+from openscm.core import (
+    ParameterAggregatedError,
+    ParameterReadonlyError,
+    ParameterSet,
+    ParameterType,
+    ParameterTypeError,
+    ParameterWrittenError,
+    RegionAggregatedError,
+)
+import pytest
+
+parameterset = ParameterSet()
+
+
+def test_region():
+    region_deu = parameterset._get_or_create_region(("DEU",))
+    assert region_deu.name == "DEU"
+
+    region_ber = parameterset._get_or_create_region(("DEU", "BER"))
+    assert region_ber.name == "BER"
+    assert region_ber.parent == region_deu
+
+    region_deu.attempt_aggregate()
+    with pytest.raises(RegionAggregatedError):
+        parameterset._get_or_create_region(("DEU", "BRB"))
+
+
+def test_parameter():
+    region_ber = parameterset._get_or_create_region(("DEU", "BER"))
+
+    with pytest.raises(ValueError):
```

Comment:
what message do we expect here?

Reply:
Right now, there is no message, should I include one such as "No region name given"?

Reply:
yes please, you can test with 

```python
error_msg = re.escape("Expected error msg")
with pytest.raises(ValueError, match=error_msg):
   code to run
```

Reply:
done